src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -73,3 +74,31 @@ def serialize(self, obj, attrs, user):
         data["projects"] = sorted(attrs["projects"])
         data["excludedProjects"] = sorted(attrs.get("excludedProjects", []))
         return data
+
+
+class CombinedRuleSerializer(Serializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        results = super(CombinedRuleSerializer, self).get_attrs(item_list, user)
+
+        alert_rules = serialize([x for x in item_list if isinstance(x, AlertRule)], user=user)
+        rules = serialize([x for x in item_list if isinstance(x, Rule)], user=user)
+
+        for item in item_list:
+            if isinstance(item, AlertRule):
+                results[item] = alert_rules[0]
```

Comment:
You can do `.pop(0)` here. Handling things like this is mildly inefficient, but it doesn't matter for such a small list of items. You could improve it by reversing the serialized lists and just popping off the end though.

Reply:
I'll look into reversing for fun and minor gainz.

But yeah, I'll make this a one liner with pop regardless.